src/config.rs

```diff
@@ -163,6 +190,65 @@ impl Config {
             None
         }
     }
+    pub(crate) fn discourse(&self) -> Option<Discourse> {
+        if let (Some(key), Some(user)) = (&self.discourse_api_key, &self.discourse_api_user) {
+            Some(Discourse::new(
+                "https://internals.rust-lang.org".to_owned(),
+                user.clone(),
+                key.clone(),
+            ))
+        } else {
+            None
+        }
+    }
+
+    pub(crate) fn blog_contents(
+        &self,
+        release: &str,
+        archive_date: &str,
+        for_blog: bool,
+        internals_url: Option<&str>,
+    ) -> Option<String> {
+        let human_date = self.scheduled_release_date.as_ref()?;
+        let release_notes_url = self.release_notes_url.as_ref()?;
+        let internals = internals_url
+            .map(|url| format!("You can leave feedback on the [internals thread]({url})."))
+            .unwrap_or_default();
+        let prefix = if for_blog {
+            format!(
+                r#"---
+layout: post
+title: "{} pre-release testing"
+author: promote-release
```

Comment:
```suggestion
author: release automation
```

Since this is public-facing I'd use a more friendly name :slightly_smiling_face: